From: pi-286-games maintainers
Subject: [PATCH] configure: do not require libpng without screenshots

The Pi 1 build deliberately removes capture support. DOSBox-X's configure
script nevertheless made libpng mandatory even with --disable-screenshots.
Avoid that unnecessary cross-sysroot dependency while retaining the existing
fatal check whenever screenshots are requested.

---
 configure.ac | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/configure.ac b/configure.ac
index 5f67c3d..e3480e1 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1154,17 +1154,17 @@ AH_TEMPLATE(C_LIBPNG,[Define to 1 if you have libpng])
 AH_TEMPLATE(C_SSHOT,[Define to 1 to enable screenshots, requires libpng])
 AC_ARG_ENABLE(screenshots,AC_HELP_STRING([--disable-screenshots],[Disable screenshots and movie recording]),,enable_screenshots=yes)
 AC_MSG_CHECKING([whether screenshots will be enabled])
-if test x$have_png_lib = xyes -a x$have_png_h = xyes ; then
-    LIBS="$LIBS -lpng -lz"
-    AC_DEFINE(C_LIBPNG,1)
-    if test x$enable_screenshots = xyes; then 
+if test x$enable_screenshots = xyes; then
+    if test x$have_png_lib = xyes -a x$have_png_h = xyes ; then
+        LIBS="$LIBS -lpng -lz"
+        AC_DEFINE(C_LIBPNG,1)
         AC_DEFINE(C_SSHOT,1)
         AC_MSG_RESULT([yes])
     else
-        AC_MSG_RESULT([no])
+        AC_MSG_ERROR([Can't find libpng, install or try building the in-tree code])
     fi
 else
-    AC_MSG_ERROR([Can't find libpng, install or try building the in-tree code])
+    AC_MSG_RESULT([no])
 fi

 dnl FEATURE: Whether to use libpcap, and enable ethernet pass-through
